ccl/changefeedccl: TestChangefeedCursor failed

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/8439ffaf-b90b-4520-a3a9-b142e6c3bc5d?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [561a43d20ce870dbe0ad12fc3e51e3bee49c60e7](https://github.com/cockroachdb/cockroach/commits/561a43d20ce870dbe0ad12fc3e51e3bee49c60e7):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor2790023106
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1090: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor2790023106
--- FAIL: TestChangefeedCursor (39.71s)
=== RUN   TestChangefeedCursor/kafka
    helpers_test.go:1043: making server as system tenant
    helpers_test.go:1119: making kafka feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:1073: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:363
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1073
        	            				pkg/ccl/changefeedccl/helpers_test.go:1291
        	            				pkg/ccl/changefeedccl/helpers_test.go:1326
        	Error:      	Received unexpected error:
        	            	operation "kafka.Next-1069712999854899201" timed out after 30s (given timeout 30s): context deadline exceeded
        	            	(1) operation "kafka.Next-1069712999854899201" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/kafka
--- FAIL: TestChangefeedCursor/kafka (39.71s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/1ac2dcea-112b-4cae-8484-ef6b8661fbb1?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [a70010142031ac4161318994cdb806b003b48cd8](https://github.com/cockroachdb/cockroach/commits/a70010142031ac4161318994cdb806b003b48cd8):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor1857891947
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1090: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor1857891947
--- FAIL: TestChangefeedCursor (359.11s)
=== RUN   TestChangefeedCursor/cloudstorage
    helpers_test.go:1040: making server as secondary tenant
    helpers_test.go:1119: making cloudstorage feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:1073: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:363
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1073
        	            				pkg/ccl/changefeedccl/helpers_test.go:1291
        	            				pkg/ccl/changefeedccl/helpers_test.go:1326
        	Error:      	Received unexpected error:
        	            	operation "cloudfeed.Next-1070612338378670081" timed out after 5m0.01s (given timeout 5m0s): context deadline exceeded
        	            	(1) operation "cloudfeed.Next-1070612338378670081" timed out after 5m0.01s (given timeout 5m0s)
        	            	Wraps: (2) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/cloudstorage
--- FAIL: TestChangefeedCursor/cloudstorage (359.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/08eef9f5-3d04-41a7-a82e-1333b0b0d501?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [6806299943a595425030f0ad3c54ccc680c89842](https://github.com/cockroachdb/cockroach/commits/6806299943a595425030f0ad3c54ccc680c89842):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor2978879756
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor2978879756
--- FAIL: TestChangefeedCursor (48.58s)
=== RUN   TestChangefeedCursor/webhook
    helpers_test.go:1047: making server as system tenant
    helpers_test.go:1123: making webhook feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "webhook.Next-1070847306416521217" timed out after 30s (given timeout 30s): context deadline exceeded
        	            	(1) operation "webhook.Next-1070847306416521217" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/webhook
--- FAIL: TestChangefeedCursor/webhook (48.58s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/45fc0a6f-c8de-4b24-9ff0-07cb2acbcdc3?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [6806299943a595425030f0ad3c54ccc680c89842](https://github.com/cockroachdb/cockroach/commits/6806299943a595425030f0ad3c54ccc680c89842):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor1244671508
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor1244671508
--- FAIL: TestChangefeedCursor (319.84s)
=== RUN   TestChangefeedCursor/cloudstorage
    helpers_test.go:1044: making server as secondary tenant
    helpers_test.go:1123: making cloudstorage feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1071159076438016001" timed out after 5m0.001s (given timeout 5m0s): operation "cloudfeed.Next-1071159076438016001" timed out after 5m0.001s (given timeout 5m0s): context deadline exceeded
        	            	(1) operation "withTimeout-1071159076438016001" timed out after 5m0.001s (given timeout 5m0s)
        	            	Wraps: (2) operation "cloudfeed.Next-1071159076438016001" timed out after 5m0.001s (given timeout 5m0s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/cloudstorage
--- FAIL: TestChangefeedCursor/cloudstorage (319.83s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/de42fd27-e556-4b20-981c-45b568466780?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [1a847b18290d439c62ff739ebdae00c851d5442e](https://github.com/cockroachdb/cockroach/commits/1a847b18290d439c62ff739ebdae00c851d5442e):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor155034015
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor155034015
--- FAIL: TestChangefeedCursor (47.48s)
=== RUN   TestChangefeedCursor/kafka
    helpers_test.go:1047: making server as system tenant
    helpers_test.go:1123: making kafka feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1072551771987902465" timed out after 30s (given timeout 30s): operation "kafka.Next-1072551771987902465" timed out after 30s (given timeout 30s): context deadline exceeded
        	            	(1) operation "withTimeout-1072551771987902465" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) operation "kafka.Next-1072551771987902465" timed out after 30s (given timeout 30s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/kafka
--- FAIL: TestChangefeedCursor/kafka (47.48s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/047f7ccb-1b1a-4d77-8c15-754d0986846b?testReportRun=2&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [a085c926b3ddeabb029d4ff5135d2165676ae7a2](https://github.com/cockroachdb/cockroach/commits/a085c926b3ddeabb029d4ff5135d2165676ae7a2):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor1036761443
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor1036761443
--- FAIL: TestChangefeedCursor (335.88s)
=== RUN   TestChangefeedCursor/webhook
    helpers_test.go:1044: making server as secondary tenant
    helpers_test.go:1123: making webhook feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1073135388374040577" timed out after 5m0s (given timeout 5m0s): operation "webhook.Next-1073135388374040577" timed out after 5m0s (given timeout 5m0s): context deadline exceeded
        	            	(1) operation "withTimeout-1073135388374040577" timed out after 5m0s (given timeout 5m0s)
        	            	Wraps: (2) operation "webhook.Next-1073135388374040577" timed out after 5m0s (given timeout 5m0s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/webhook
--- FAIL: TestChangefeedCursor/webhook (335.87s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/5bd55438-7c0e-4b8c-9089-eb9330cf6969?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [efbd1971046b006f4ee903c146586e9ea8e6d1e4](https://github.com/cockroachdb/cockroach/commits/efbd1971046b006f4ee903c146586e9ea8e6d1e4):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor1905220500
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor1905220500
--- FAIL: TestChangefeedCursor (43.99s)
=== RUN   TestChangefeedCursor/webhook
    helpers_test.go:1047: making server as system tenant
    helpers_test.go:1123: making webhook feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1074269635280732161" timed out after 30s (given timeout 30s): operation "webhook.Next-1074269635280732161" timed out after 30s (given timeout 30s): context deadline exceeded
        	            	(1) operation "withTimeout-1074269635280732161" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) operation "webhook.Next-1074269635280732161" timed out after 30s (given timeout 30s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/webhook
--- FAIL: TestChangefeedCursor/webhook (43.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/3f049df4-26bb-4a47-9717-00db24b8b992?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [e42de35ba24e5ef3086e625a2654fd98d005f894](https://github.com/cockroachdb/cockroach/commits/e42de35ba24e5ef3086e625a2654fd98d005f894):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor995978041
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor995978041
--- FAIL: TestChangefeedCursor (324.37s)
=== RUN   TestChangefeedCursor/cloudstorage
    helpers_test.go:1044: making server as secondary tenant
    helpers_test.go:1123: making cloudstorage feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1075992170702929921" timed out after 5m0.007s (given timeout 5m0s): stopping job due to TestFeed timeout
        	            	(1) operation "withTimeout-1075992170702929921" timed out after 5m0.007s (given timeout 5m0s)
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func3
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2974
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func1
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2963
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (3) stopping job due to TestFeed timeout
        	            	Error types: (1) *timeutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestChangefeedCursor/cloudstorage
--- FAIL: TestChangefeedCursor/cloudstorage (324.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/5f8efff1-5354-4c78-a8d6-9b28c33e7fb4?testReportRun=2&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [4f26eda2e687eea330e61f62f5dfc069070a65cd](https://github.com/cockroachdb/cockroach/commits/4f26eda2e687eea330e61f62f5dfc069070a65cd):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor3598587751
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor3598587751
--- FAIL: TestChangefeedCursor (46.92s)
=== RUN   TestChangefeedCursor/kafka
    helpers_test.go:1047: making server as system tenant
    helpers_test.go:1123: making kafka feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1076229163300388865" timed out after 30s (given timeout 30s): stopping job due to TestFeed timeout
        	            	(1) operation "withTimeout-1076229163300388865" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func3
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2974
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func1
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2963
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (3) stopping job due to TestFeed timeout
        	            	Error types: (1) *timeutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestChangefeedCursor/kafka
--- FAIL: TestChangefeedCursor/kafka (46.92s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/5b01f148-185f-48c1-8158-4acb22021fdc?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [66bff6a1827cddb8a7e9c396a0f76f311c0ddc65](https://github.com/cockroachdb/cockroach/commits/66bff6a1827cddb8a7e9c396a0f76f311c0ddc65):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor3606026022
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor3606026022
--- FAIL: TestChangefeedCursor (320.36s)
=== RUN   TestChangefeedCursor/pubsub
    helpers_test.go:1044: making server as secondary tenant
    helpers_test.go:1123: making pubsub feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1076281107040600065" timed out after 5m0.001s (given timeout 5m0s): operation "pubsub.Next-1076281107040600065" timed out after 5m0.001s (given timeout 5m0s): context deadline exceeded
        	            	(1) operation "withTimeout-1076281107040600065" timed out after 5m0.001s (given timeout 5m0s)
        	            	Wraps: (2) operation "pubsub.Next-1076281107040600065" timed out after 5m0.001s (given timeout 5m0s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/pubsub
--- FAIL: TestChangefeedCursor/pubsub (320.36s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/ff503aec-a5b7-44fe-9e8e-c8fe9999107f?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [b216877de8156c9907ecb5ac5b9129104f4fac31](https://github.com/cockroachdb/cockroach/commits/b216877de8156c9907ecb5ac5b9129104f4fac31):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor2717174544
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor2717174544
--- FAIL: TestChangefeedCursor (47.12s)
=== RUN   TestChangefeedCursor/kafka
    helpers_test.go:1047: making server as system tenant
    helpers_test.go:1123: making kafka feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1077082350080524289" timed out after 30s (given timeout 30s): operation "kafka.Next-1077082350080524289" timed out after 30s (given timeout 30s): context deadline exceeded
        	            	(1) operation "withTimeout-1077082350080524289" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) operation "kafka.Next-1077082350080524289" timed out after 30s (given timeout 30s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/kafka
--- FAIL: TestChangefeedCursor/kafka (47.12s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/f923e8ba-6173-432c-a880-4baa49a8c5c7?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [8fed6e8481da1373020ab2cc95d222df02f61721](https://github.com/cockroachdb/cockroach/commits/8fed6e8481da1373020ab2cc95d222df02f61721):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor665991671
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor665991671
--- FAIL: TestChangefeedCursor (39.60s)
=== RUN   TestChangefeedCursor/kafka
    helpers_test.go:1047: making server as system tenant
    helpers_test.go:1123: making kafka feed factory
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    testfeed_test.go:322: creating external connection
    testfeed_test.go:325: ran create external connection
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:268
        	            				pkg/ccl/changefeedccl/helpers_test.go:367
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1295
        	            				pkg/ccl/changefeedccl/helpers_test.go:1330
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1077359080270233601" timed out after 30s (given timeout 30s): stopping job due to TestFeed timeout
        	            	(1) operation "withTimeout-1077359080270233601" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func3
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2974
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func1
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2963
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (3) stopping job due to TestFeed timeout
        	            	Error types: (1) *timeutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestChangefeedCursor/kafka
--- FAIL: TestChangefeedCursor/kafka (39.59s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/458f6de1-3236-4c30-bae3-0c0f8b15cbe8?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [2c99066211911e14c7519ec47a5c021488a1fe71](https://github.com/cockroachdb/cockroach/commits/2c99066211911e14c7519ec47a5c021488a1fe71):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor2446235337
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor2446235337
--- FAIL: TestChangefeedCursor (47.04s)
=== RUN   TestChangefeedCursor/kafka
    helpers_test.go:1238: making server as system tenant
    helpers_test.go:1314: making kafka feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1080196273554391041" timed out after 30s (given timeout 30s): stopping job due to TestFeed timeout
        	            	(1) operation "withTimeout-1080196273554391041" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func3
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:3000
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func1
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2989
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (3) stopping job due to TestFeed timeout
        	            	Error types: (1) *timeutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestChangefeedCursor/kafka
--- FAIL: TestChangefeedCursor/kafka (47.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/38dd3134-44c6-454a-8ff9-f8890d0d41d0?testReportRun=1&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [28e1b226dfb3ceb38af3214d5b9e415d8e0c3c92](https://github.com/cockroachdb/cockroach/commits/28e1b226dfb3ceb38af3214d5b9e415d8e0c3c92):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor1435645851
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor1435645851
--- FAIL: TestChangefeedCursor (38.31s)
=== RUN   TestChangefeedCursor/webhook
    helpers_test.go:1238: making server as system tenant
    helpers_test.go:1314: making webhook feed factory
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1081323145876013057" timed out after 30s (given timeout 30s): stopping job due to TestFeed timeout
        	            	(1) operation "withTimeout-1081323145876013057" timed out after 30s (given timeout 30s)
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func3
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:3000
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func1
        	            	  | 	pkg/ccl/changefeedccl/testfeed_test.go:2989
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (3) stopping job due to TestFeed timeout
        	            	Error types: (1) *timeutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestChangefeedCursor/webhook
--- FAIL: TestChangefeedCursor/webhook (38.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

ccl/changefeedccl.TestChangefeedCursor [failed](https://tanzanite.cluster.engflow.com/invocations/default/d19e6149-c3c0-4691-8daa-aad1dc551867?testReportRun=3&testReportShard=11&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [93519aef3385e7dca7b2cda8c528e6fa3027f6e9](https://github.com/cockroachdb/cockroach/commits/93519aef3385e7dca7b2cda8c528e6fa3027f6e9):


```
=== RUN   TestChangefeedCursor
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestChangefeedCursor3305645680
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:1091: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedCursor3305645680
--- FAIL: TestChangefeedCursor (338.98s)
=== RUN   TestChangefeedCursor/cloudstorage
    helpers_test.go:1235: making server as secondary tenant
    helpers_test.go:1314: making cloudstorage feed factory
    testfeed_test.go:325: creating external connection
    testfeed_test.go:328: ran create external connection
    testfeed_test.go:325: creating external connection
    testfeed_test.go:328: ran create external connection
    testfeed_test.go:325: creating external connection
    testfeed_test.go:328: ran create external connection
    testfeed_test.go:325: creating external connection
    testfeed_test.go:328: ran create external connection
    changefeed_test.go:1074: 
        	Error Trace:	pkg/ccl/changefeedccl/helpers_test.go:273
        	            				pkg/ccl/changefeedccl/helpers_test.go:447
        	            				pkg/ccl/changefeedccl/changefeed_test.go:1074
        	            				pkg/ccl/changefeedccl/helpers_test.go:1486
        	            				pkg/ccl/changefeedccl/helpers_test.go:1521
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-1081946843895955457" timed out after 5m0.007s (given timeout 5m0s): operation "cloudfeed.Next-1081946843895955457" timed out after 5m0.007s (given timeout 5m0s): context deadline exceeded
        	            	(1) operation "withTimeout-1081946843895955457" timed out after 5m0.007s (given timeout 5m0s)
        	            	Wraps: (2) operation "cloudfeed.Next-1081946843895955457" timed out after 5m0.007s (given timeout 5m0s)
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *timeutil.TimeoutError (2) *timeutil.TimeoutError (3) context.deadlineExceededError
        	Test:       	TestChangefeedCursor/cloudstorage
--- FAIL: TestChangefeedCursor/cloudstorage (338.97s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=11</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedCursor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -1029,62 +1029,70 @@ func TestChangefeedCursor(t *testing.T) {
 		require.True(t, beforeInsert.Less(insertTimestamp) && insertTimestamp.Less(tsLogical) && tsLogical.Less(s.Server.Clock().Now()),
 			fmt.Sprintf("beforeInsert: %s, insertTimestamp: %s, tsLogical: %s", beforeInsert, insertTimestamp, tsLogical))
 
-		// The below function is currently used to test negative timestamp in cursor i.e of the form
-		// "-3us".
-		// Using this function we can calculate the difference with the time that was before
-		// the insert statement, which is set as the new cursor value inside createChangefeedJobRecord
-		calculateCursor := func(currentTime *hlc.Timestamp) string {
-			//  Should convert to microseconds as that is the maximum precision we support
-			diff := (beforeInsert.WallTime - currentTime.WallTime) / 1000
-			diffStr := strconv.FormatInt(diff, 10) + "us"
-			return diffStr
-		}
+		t.Run("negative cursor", func(t *testing.T) {
+			// The below function is currently used to test negative timestamp in cursor i.e of the form
+			// "-3us".
+			// Using this function we can calculate the difference with the time that was before
+			// the insert statement, which is set as the new cursor value inside createChangefeedJobRecord
+			calculateCursor := func(currentTime *hlc.Timestamp) string {
+				//  Should convert to microseconds as that is the maximum precision we support
+				diff := (beforeInsert.WallTime - currentTime.WallTime) / 1000
+				diffStr := strconv.FormatInt(diff, 10) + "us"
+				return diffStr
+			}
 
-		knobs := s.TestingKnobs.DistSQL.(*execinfra.TestingKnobs).Changefeed.(*TestingKnobs)
-		knobs.OverrideCursor = calculateCursor
+			knobs := s.TestingKnobs.DistSQL.(*execinfra.TestingKnobs).Changefeed.(*TestingKnobs)
+			knobs.OverrideCursor = calculateCursor
 
-		// The "-3 days" is a placeholder here - it will be replaced with actual difference
-		// in createChangefeedJobRecord
-		fooInterval := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, "-3 days")
-		defer closeFeed(t, fooInterval)
-		assertPayloads(t, fooInterval, []string{
-			`foo: [1]->{"after": {"a": 1, "b": "before"}}`,
-			`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+			// The "-3 days" is a placeholder here - it will be replaced with actual difference
+			// in createChangefeedJobRecord
+			fooInterval := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, "-3 days")
+			defer closeFeed(t, fooInterval)
+			assertPayloads(t, fooInterval, []string{
+				`foo: [1]->{"after": {"a": 1, "b": "before"}}`,
+				`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+			})
+
+			// We do not need to override for the remaining cases
+			knobs.OverrideCursor = nil
 		})
 
-		// We do not need to override for the remaining cases
-		knobs.OverrideCursor = nil
+		t.Run("decimal cursor", func(t *testing.T) {
+			fooLogical := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, eval.TimestampToDecimalDatum(tsLogical).String())
+			defer closeFeed(t, fooLogical)
+			assertPayloads(t, fooLogical, []string{
+				`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+			})
 
-		fooLogical := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, eval.TimestampToDecimalDatum(tsLogical).String())
-		defer closeFeed(t, fooLogical)
-		assertPayloads(t, fooLogical, []string{
-			`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+			// Check that the cursor is properly hooked up to the job statement
+			// time. The sinkless tests currently don't have a way to get the
+			// statement timestamp, so only verify this for enterprise.
+			if e, ok := fooLogical.(cdctest.EnterpriseTestFeed); ok {
+				var bytes []byte
+				sqlDB.QueryRow(t, jobutils.JobPayloadByIDQuery, e.JobID()).Scan(&bytes)
+				var payload jobspb.Payload
+				require.NoError(t, protoutil.Unmarshal(bytes, &payload))
+				require.Equal(t, tsLogical, payload.GetChangefeed().StatementTime)
+			}
 		})
 
-		nanosStr := strconv.FormatInt(tsClock.UnixNano(), 10)
-		fooNanosStr := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, nanosStr)
-		defer closeFeed(t, fooNanosStr)
-		assertPayloads(t, fooNanosStr, []string{
-			`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+		t.Run("nanos cursor", func(t *testing.T) {
+			nanosStr := strconv.FormatInt(tsClock.UnixNano(), 10)
+			fooNanosStr := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, nanosStr)
+			defer closeFeed(t, fooNanosStr)
+			assertPayloads(t, fooNanosStr, []string{
+				`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+			})
 		})
 
-		timeStr := tsClock.Format(`2006-01-02 15:04:05.999999`)
-		fooString := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, timeStr)
-		defer closeFeed(t, fooString)
-		assertPayloads(t, fooString, []string{
-			`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+		t.Run("datetime cursor", func(t *testing.T) {
+			timeStr := tsClock.Format(`2006-01-02 15:04:05.999999`)
+			fooString := feed(t, f, `CREATE CHANGEFEED FOR foo WITH cursor=$1`, timeStr)
+			defer closeFeed(t, fooString)
+			assertPayloads(t, fooString, []string{
+				`foo: [2]->{"after": {"a": 2, "b": "after"}}`,
+			})
 		})
-
-		// Check that the cursor is properly hooked up to the job statement
-		// time. The sinkless tests currently don't have a way to get the
-		// statement timestamp, so only verify this for enterprise.
-		if e, ok := fooLogical.(cdctest.EnterpriseTestFeed); ok {
-			var bytes []byte
-			sqlDB.QueryRow(t, jobutils.JobPayloadByIDQuery, e.JobID()).Scan(&bytes)
-			var payload jobspb.Payload
-			require.NoError(t, protoutil.Unmarshal(bytes, &payload))
-			require.Equal(t, tsLogical, payload.GetChangefeed().StatementTime)
-		}
 	}
 
 	cdcTest(t, testFn)

